### PR TITLE
feat(api): Migrate GCP Pub Sub alert channel to API v2

### PR DESCRIPTION
**Pull request by vatasha**
## Summary
Migrate GCP pub sub alert channel to API v2 as a part of a greater effort to manage alert channels at the org level. 

## How did you test this change?

There are new tests and I updated existing ones. 

## Issue
https://lacework.atlassian.net/browse/ALLY-641?atlOrigin=eyJpIjoiMDZiZGNlNjM0ZDdkNDNkOGFkNDdiNGYzYjZhN2IxYmEiLCJwIjoiaiJ9

### CODE REVIEW

File: api/alert_channels_gcp_pub_sub.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// UpdateGcpPubSub updates a single instance of GCP Pub Sub integration on the Lacework server
+func (svc *AlertChannelsService) UpdateGcpPubSub(data AlertChannel) (response GcpPubSubAlertChannelResponseV2, err error) {
+	err = svc.update(data.ID(), data, &response)
+	return
+}
+
+type GcpPubSubDataV2 struct {
+	Credentials   GcpPubSubCredentials `json:"credentials"`
+	IssueGrouping string               `json:"issueGrouping"`
+	ProjectId     string               `json:"ProjectId"`
```
Comment (dmurray-lacework) [inline api/alert_channels_gcp_pub_sub.go:36]: Does this need to be lower case?
```suggestion
	ProjectId     string               `json:"projectId"`
```

Comment (afiune) [reply api/alert_channels_gcp_pub_sub.go:36]: Our unit tests should have caught this but, we are not testing correctly, we are missing a few fields so please, lets add assertions for all these fields within the DataV2 struct.

File: api/alert_channels_gcp_pub_sub_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
+	assert.Equal(t, intgGUID, response.Data.IntgGuid)
+	assert.Equal(t, "integration_name", response.Data.Name)
+	assert.True(t, response.Data.State.Ok)
+	assert.Equal(t, "vatasha.white@lacework.net", response.Data.Data.Credentials.ClientEmail)
+	assert.Equal(t, "fake-client-id", response.Data.Data.Credentials.ClientId)
+	assert.Equal(t, "fake-private-key", response.Data.Data.Credentials.PrivateKey)
+	assert.Equal(t, "fake-private-key-id", response.Data.Data.Credentials.PrivateKeyId)
+	assert.Equal(t, "fake-project-id", response.Data.Data.ProjectId)
```
Comment (afiune) [inline api/alert_channels_gcp_pub_sub_test.go:65]:
```suggestion
                       assert.Contains(t, body, "projectId\":\"fake-project-id", "missing project id")
```

File: api/alert_channels_gcp_pub_sub.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+type GcpPubSubAlertChannelResponseV2 struct {
+	Data GcpPubSubAlertChannelV2 `json:"data"`
+}
+
+type GcpPubSubCredentials struct {
+	ClientEmail  string `json:"clientEmail"`
+	ClientId     string `json:"clientId"`
+	PrivateKey   string `json:"privateKey"`
+	PrivateKeyId string `json:"privateKeyId"`
```
Comment (afiune) [inline api/alert_channels_gcp_pub_sub.go:53]: I still think we should call these two fields ID capitalized, but it is a nitpick 
```suggestion
	ClientID     string `json:"clientId"`
	PrivateKey   string `json:"privateKey"`
	PrivateKeyID string `json:"privateKeyId"`
```

**Review by afiune — APPROVED** [review]
tenor-219325769

**Pull request merged**